src/Composer/ComposerWidget.vala

```diff
@@ -490,7 +490,26 @@ public class Mail.ComposerWidget : Gtk.Grid {
     }
 
     private void on_discard () {
-        discarded ();
+        var discard_dialog = new Granite.MessageDialog.with_image_from_icon_name (
+            _("Delete This Draft?"), // choice of 'Delete' over 'Discard' for clarity
+            _("Are you sure you want to delete this draft permanently?"),
+            "dialog-error",
+            Gtk.ButtonsType.NONE
+        );
+        discard_dialog.transient_for = get_toplevel () as Gtk.Window;
+
+        discard_dialog.add_button (_("No"), Gtk.ResponseType.CANCEL);
```

Comment:
Action buttons should never be "Yes", "No", or "OK". They should always describe an explicit action. In this case, it would be much better to use "Cancel"
```suggestion
        discard_dialog.add_button (_("Cancel"), Gtk.ResponseType.CANCEL);
```